### PR TITLE
TYP: enforce tighter inputs on SingleBlockManager

### PR DESCRIPTION
**Pull request by jbrockmendel**
there is currently only one place where we pass a list of indexes instead of a single index, so just fixed that and were good to go.

### CODE REVIEW

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -1649,6 +1640,7 @@ def concat(self, to_concat, new_axis: Index) -> "SingleBlockManager":
         -------
         SingleBlockManager
         """
+        assert isinstance(new_axis, Index), new_axis
```
Comment (jorisvandenbossche) [inline pandas/core/internals/managers.py]: this is already asserted in the SingleBlockManager constructor, so is not needed here?

File: pandas/core/internals/managers.py [hunk 2]
```diff
@@ -1515,25 +1515,14 @@ class SingleBlockManager(BlockManager):
     def __init__(
         self,
         block: Block,
-        axis: Union[Index, List[Index]],
+        axis: Index,
         do_integrity_check: bool = False,
         fastpath: bool = False,
```
Comment (jorisvandenbossche) [inline pandas/core/internals/managers.py]: `fastpath` has no effect anymore, now? It is still being used in places internally?

**Comment by jreback** [comment]
lgtm. rebase and @jorisvandenbossche has some comments, ping on green.

**Comment by jbrockmendel** [comment]
@jorisvandenbossche i think ive addressed your comments, can you confirm?

**Comment by jorisvandenbossche** [comment]
Yes, thanks.

One more thing, though. For `Block`, we actually deprecated `fastpath` first instead of directly removing (only the passing of the keyword, I mean, not any functionality, https://github.com/pandas-dev/pandas/pull/19265). I assume SingleBlockManager will be used less in practice, but it's also not much work to still accept fastpath and raise a warning for it for now.

**Comment by jbrockmendel** [comment]
> deprecated `fastpath` first instead of directly removing

Sure

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**